### PR TITLE
Update to v0.0.2 with related automated workflows

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This PR prepares for the v0.0.2 release of CytoTable and adds various related enhancements such as metadata and automated workflows. On merging these changes I expect and will publish the draft release automatically created, which (on release draft publishing) then will publish built artifacts to PyPI.

Notable changes outside this PR in preparation for releases:
- Added labels `release-patch`, `release-minor` and `release-major` to assist with automated draft releases provided through `release-drafter`.
- Prepared PyPI CytoTable management to use [Trusted Publishing](https://docs.pypi.org/trusted-publishers/) methods.

## What is the nature of your change?

- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] This change requires a documentation update.

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -20,11 +20,10 @@ authors:
...
     family-names: Way
     orcid: 'https://orcid.org/0000-0002-0503-9348'
-repository-code: 'https://github.com/cytomining/pycytominer-transform'
-url: 'https://cytomining.github.io/pycytominer-transform/'
+repository-code: 'https://github.com/cytomining/CytoTable'
+url: 'https://cytomining.github.io/CytoTable/'
 abstract: >-
-  Transform single cell morphology data for image-based profiling applications and analyses.
-  readouts with Pycytominer.
+  Transform data for processing image-based profiling readouts with Pycytominer and other Cytomining tools.
```
Comment (gwaybio) [inline CITATION.cff:26]:
```suggestion
  Transform CellProfiler and DeepProfiler data for processing image-based profiling readouts with Pycytominer and other Cytomining tools.
```

File: docs/source/contributing.md [hunk 2]
```diff
@@ -171,3 +171,22 @@ Documentation is automatically published to a docsite via [GitHub Actions](https
...
+We publish source code by using [GitHub Releases](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases) available [here](https://github.com/cytomining/CytoTable/releases).
+We publish Python packages through the [Python Packaging Index (PyPI)](https://pypi.org/) available [here](https://pypi.org/project/cytotable/).
+
+### Publishing Process
+
+There are several manual and automated steps involved with publishing CytoTable releases.
+See below for an overview of how this works.
+
+1. Prepare a release in a code contribution which utilizes the command [`poetry version ...`](https://python-poetry.org/docs/cli/#version) to update the version (this updates `pyproject.toml` with automatically incremented versions under `version = "..."`).
+1. Open a pull request and use a repository label for `release-<semver release type>` to label the pull request for visibility with [`release-drafter`](https://github.com/release-drafter/release-drafter).
```
Comment (gwaybio) [inline docs/source/contributing.md:189]: Is there an example PR you can point to?

File: pyproject.toml [hunk 3]
```diff
@@ -1,10 +1,14 @@
 [tool.poetry]
 name = "CytoTable"
-version = "0.0.1"
-description = "Transform data for processing image-based profiling readouts with Pycytominer."
+version = "0.0.2"
+description = "Transform data for processing image-based profiling readouts with Pycytominer and other Cytomining tools."
```
Comment (gwaybio) [inline pyproject.toml:4]:
```suggestion
description = "Transform CellProfiler and DeepProfiler data for processing image-based profiling readouts with Pycytominer and other Cytomining tools."
```

**Review by gwaybio — APPROVED** [review]

**Review by falquaddoomi — APPROVED** [review]
Cool to see how much of the release and publishing process can be automated with GH actions. Nice work!

File: CITATION.cff (hunk 1 above)
Comment (d33bs) [reply CITATION.cff:26]: Thank you for this! Committing into an upcoming batch.

File: docs/source/contributing.md (hunk 2 above)
Comment (d33bs) [reply docs/source/contributing.md:189]: Thanks, great point! I'll add a reference to this PR as an example of this occurring within CytoTable.

**Comment by d33bs** [comment]
Thank you @gwaybio and @falquaddoomi for your reviews! I've addressed all comments with subsequent updates. Merging this in now.

**Pull request merged**